src/lib/semaphore.js

```diff
@@ -0,0 +1,29 @@
+export default class Semaphore {
```

Comment:
Is there any way to write a test for this?

Reply:
sure, I will add a test